### PR TITLE
Fix code style and other minor corrections after implementing allow_no_password.

### PR DESCRIPTION
**Pull request by vitlibar**
Changelog category:
- Not for changelog

### CODE REVIEW

File: src/Access/AccessControl.cpp [hunk 1]
```diff
@@ -447,23 +450,35 @@ void AccessControl::setCustomSettingsPrefixes(const String & comma_separated_pre
     setCustomSettingsPrefixes(prefixes);
 }
 
-void AccessControl::setPlaintextPasswordSetting(bool allow_plaintext_password_)
+bool AccessControl::isSettingNameAllowed(const std::string_view & setting_name) const
```
Comment (antonio2368) [inline src/Access/AccessControl.cpp:453]:
```suggestion
bool AccessControl::isSettingNameAllowed(const std::string_view setting_name) const
```
`std::string_view` is just pointer and size so it's better to copy them than add an extra indirection with reference.

**Comment by vitlibar** [comment]
Test failures are not related to this PR's changes

**Comment by arthurpassos** [comment]
@vitlibar as far as I understand, one of the things this PR does is shutdown ClickHouse in case a user with `plaintext_password` exists and `allow_plaintext_password` is set to false. If that is the case, does the check for `allow_plaintext_password` upon authentication even make sense? It seems like that code would be unreacheable unless I am missing something.

I am asking because I am working on adding support to multiple authentication methods and that check is kind of getting in the way.

https://github.com/ClickHouse/ClickHouse/blob/4a30ebd8d612b9c7c041904022cd46c2454e606f/src/Access/IAccessStorage.cpp#L529

**Comment by arthurpassos** [comment]
> @vitlibar as far as I understand, one of the things this PR does is shutdown ClickHouse in case a user with `plaintext_password` exists and `allow_plaintext_password` is set to false. If that is the case, does the check for `allow_plaintext_password` upon authentication even make sense? It seems like that code would be unreacheable unless I am missing something.
> 
> I am asking because I am working on adding support to multiple authentication methods and that check is kind of getting in the way.
> 
> https://github.com/ClickHouse/ClickHouse/blob/4a30ebd8d612b9c7c041904022cd46c2454e606f/src/Access/IAccessStorage.cpp#L529

Turns out that if the user was created through SQL when this setting did not exist or it was enabled, clickhouse doesn't shutdown. That's why this check exists.

**Pull request merged**